lib/3scale/backend/configuration.rb

```diff
@@ -60,6 +72,22 @@ def configure
         ENV['CONFIG_FILE']
       ].compact)
 
+      # Assign default values to some configuration values
+      # that might been set in the config file but their
+      # environment variables not. Also make sure
+      # that what we have is a String, just in case
+      # from the config file a type of data different than a
+      # String has been assigned to this configuration
+      # parameter
+      config.master.metrics.transactions = config.master.metrics.transactions.to_s
+      if config.master.metrics.transactions.empty?
```

Comment:
A user can assign `nil` in his config file. Is `nil` a value we should accept? If it's not, we should also check for `nil` here.

Reply:
Also, if we check for nil here, we could delete the 2 assignments above.

Reply:
Hi,

This was done intentionally in this way. When calling the .to_s method to an object that is nil the result is converted into the empty string:
```
irb(main):004:0> mystring = nil
=> nil
irb(main):005:0> mystring.to_s
=> ""
```

Reply:
Regarding the redundancy of the 2 assignments above, I agree with you and I saw this too but it seemed to me that the "defaults assignment" logic was located there among the other currently set default configuration parameters so I kept that assignments there just to be clear (even if they are repeated code).
I can remove it without any problem if we prefer to avoid code duplication in this case.